cmd/gaia/init/genesis_accts.go

```diff
@@ -86,6 +107,38 @@ func addGenesisAccount(cdc *codec.Codec, appState app.GenesisState, addr sdk.Acc
 
 	acc := auth.NewBaseAccountWithAddress(addr)
 	acc.Coins = coins
-	appState.Accounts = append(appState.Accounts, app.NewGenesisAccount(&acc))
+
+	if !vestingAmt.IsZero() {
+		var vacc auth.VestingAccount
+
+		bvacc := &auth.BaseVestingAccount{
+			BaseAccount:     &acc,
+			OriginalVesting: vestingAmt,
+			EndTime:         vestingEnd,
+		}
+
+		if bvacc.OriginalVesting.IsAllGT(acc.Coins) {
```

Comment:
Is `bvacc.OriginalVesting.Validate()` called somewhere? If not, I'd make sure it is

Reply:
You're right :+1: